src/components/form/examples/form-span-fields.tsx

```diff
@@ -0,0 +1,215 @@
+import { schema } from './schema-span-fields';
+
+/**
+ * Stretching fields in a form
+ * Sometimes, you need a field in the form to occupy several columns or the
+ * entire row, and stretch itself as wide as the form's width;
+ * disregarding the form's layout and placement of the item in the list.
+ *
+ * This could be nice for fields that require more space to provide a more
+ * decent usability.
+ *
+ * :::tip
+ * For example, a larger `textarea` is easier for the user to type in and
+ * a `slider` that has many steps is easier to interact with when it is rendered wider.
+ * :::
+ *
+ * To do so, in your `schema` you need to specify a `layout` for the field itself.
+ * The `span` number you specify will be the number of columns that the field
+ * should span through the columns of the grid.
+ *
+ * Thus the `span` value can be set to `2`, `3`, `4`, `5`, or `all`.
+ * Since we do not offer a *form layout* that has more than five columns,
+ * values higher than 5 (or higher than maximum number of columns in the form)
+ * will only force the file to be full-width, just like what `all` does.
+ *
+ * ````ts
+ * export const schema = {
+ *     …
+ *     properties: {
+ *         name: {
+ *             type: 'string',
+ *             title: 'Comment',
+ *             lime: {
+ *                 layout: {
+ *                       colSpan: 'all',
+ *                   },
+ *               },
+ *         },
+ *         …
+ *     },
+ *     …
+ * };
+ * ````
+ *
+ * ###### Dense layout (Auto reorder fields to avoid empty cells)
+ * Order of fields and number of columns that a field must span through, can
+ * consequently affect the layout of your responsive form when the screen width
+ * changes.
+ * Let's say you have a form with a 4 columned layout, and you specify that its
+ * second field must span (or stretch) through 3 columns.
+ * Now if the container's width decreases, it will forces the form to render its
+ * layout in 3 columns instead. Therefore, the second field has to jump
+ * to the next line to still be able to stretch through 3 columns.
+ * This will leave 2 empty cells in the first row, right after the first field.
+ *
+ * To avoid having these empty cells in the UI, limel-form will automatically
+ * bring up two other fields that span only 1 column, and places them in the
+ * available empty cells in the first row.
+ *
+ * However, you can disable this functionality by setting `dense` to fall in the
+ * options for the grid layout
+ *
+ * ```ts
+ * export const schema = {
+ *     type: 'object',
+ *     lime: {
+ *         layout: {
+ *             type: 'grid',
+ *             dense: false,
+ *         },
+ *     },
+ * };
+ * ```
+ *
+ * :::note
+ * Sometimes order of fields are important in the way users perceive the form.
+ * If you choose to have the default auto-reordering behavior of the forms,
+ * make sure to test yout forms layout in different screen sizes to see whether
+ * you can mitigate unwanted layout changes.
```

Comment:
```suggestion
 * Sometimes, the order of fields are important for the way users perceive the form.
 * If you choose to use the default auto-reordering behavior, make sure to test your
 * form's layout in different screen sizes to see whether you can mitigate unwanted
 * layout changes.
```

Reply:
I think this ☝️ is a very good reason for not having `dense` default to `true`. How many users of limel-form is actually going to do that?